src/runtime/methodbinder.cs

```diff
@@ -519,6 +536,13 @@ static IntPtr HandleParamsArray(IntPtr args, int arrayStart, int pyArgCount, out
                         op = Runtime.PyTuple_GetItem(args, paramIndex);
                     }
                 }
+                if (isOperator && paramIndex == 0)
+                {
+                    // After we've obtained the first argument from Python, we need to skip the first argument of the CLR
+                    // because operator method is a bound method in Python
+                    paramIndex++; // Leave the first .NET param as null (margs).
+                    parameter = pi[paramIndex];
+                }
```

Comment:
Seeing this discussion and `bool isOperator` doing control flow actually led me to think, that the design around operators would be much better if instead an `OperatorBinder` class would be created to derive from `MethodBinder`, and the control flow around branches, that use `isOperator` be refactored so that `OperatorBinder` could do its thing.

Or even better it could just preprocess arguments for or postprocess results of `TryConvertArguments`.

@christabella @tminka do you think you would have time to do that? It would be rather involved.

Reply:
I agree that it is better to remove `IsOperator` from `TryConvertArguments` and use preprocessing and postprocessing instead.  Especially since there is already some preprocessing and postprocessing going on.

Reply:
Thanks for the suggestion. When handling reverse operators in the next PR, perhaps we could do the necessary pre- and post-processing of `TryConvertArguments` and remove `isOperator` from `TryConvertArguments`, as @tminka and I just discussed.

Reply:
The next PR after this one is ready on my fork https://github.com/christabella/pythonnet/pull/1/files where we do preprocessing and postprocessing instead of passing `isOperator` into `TryConvertArguments`, so this part of the code is removed completely in that one.